File: roblox_plugin/src/Server/pauseThread.ts
export interface PauseThread {
	thread?: thread;
	stopThread: boolean;
}
export namespace PauseThread {
	export function pause(pauseThread: PauseThread) {
		pauseThread.thread = coroutine.running();
		coroutine.yield();
	}
	export function unpause(pauseThread: PauseThread) {
		if (pauseThread.thread) {
			const thread = pauseThread.thread;
			pauseThread.thread = undefined;
			if (pauseThread.stopThread) {
				coroutine.close(thread);
			} else {
				coroutine.resume(thread);
			}
		}
	}
	export function stop(pauseThread: PauseThread) {
		pauseThread.stopThread = true;
	}
}
export default PauseThread;

File: blender_roblox_sync/server/pause_thread.py
import time, threading, sys

class PauseThread(threading.Thread):
	def pause(self):
		self.paused = True
		while self.paused:
			time.sleep(0.001)
			if self.stop_thread:
				sys.exit()

	def unpause(self):
		self.paused = False

	def stop(self):
		self.stop_thread = True